static/js/controllers/contacts.js

```diff
@@ -151,6 +142,44 @@ var _ = require('underscore'),
             _query();
           }
         }
+      };
+
+      $scope.setupSearchFreetext = function() {
+        SearchFilters.freetext($scope.search);
+      };
+      $scope.setupSearchFacility = function() {
+        SearchFilters.facility(function(facilities) {
+          $scope.filters.facilities = facilities;
+          $scope.search();
+        });
+      };
+      $scope.setupSearchContactType = function() {
+        SearchFilters.contactType(function(types) {
+          $scope.filters.types = types;
+          $scope.search();
+        });
+      };
+      $scope.resetFilterModel = function() {
+        $scope.filters = {};
+        SearchFilters.reset();
+        $scope.search();
+      };
+
+      $scope.search();
+
+      $scope.$on('export', function() {
+        if ($scope.currentTab === 'contacts') {
```

Comment:
_(apologies if this is in the tests or explained elsewhere)_

How could this fail? How could this function be ran but the current tab not be contacts? Does this `$on` get setup at angular init, and so it being in contacts controller is merely a useful code sorting thing for humans and everyone's `$on`s get run every time, or does angular only have that attached when the contacts controller is active?